### PR TITLE
Roles Feature

### PR DESCRIPTION
**Pull request by aaronbrowder**
This is a new feature, found under a new main tab called "Roles", which must be enabled in Settings.

This allows demschooltools to track clerkships, committee memberships, etc. The main benefit of this (as opposed to a regular spreadsheet) is that it tracks role history, which can be used for student transcripts. In the Records sub-tab, you can enter a person's name and it will show you all the roles they ever held, with dates.

Let me know if you'd like more details on how it works.

screenshot

### CODE REVIEW

**Comment by schmave** [comment]
Looks nice! I'll double check things and get this merged on Friday.

File: modelsLibrary/app/models/RoleRecordMember.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  @ManyToOne()
+  @JsonIgnore
+  private RoleRecord record;
+
+  @ManyToOne()
+  @JoinColumn(name = "person_id")
+  @JsonIgnore
+  private Person person;
+
+  private String personName;
```
Comment (schmave) [inline modelsLibrary/app/models/RoleRecordMember.java:28]: Why do you store personName as well as the person record?

**Review by schmave — APPROVED** [review]
Looks great to me! One question below about the data model.

File: modelsLibrary/app/models/RoleRecordMember.java (hunk 1 above)
Comment (aaronbrowder) [reply modelsLibrary/app/models/RoleRecordMember.java:28]: This allows adding a person who is not in the system. For example, if a parent is a member of a committee, there would be no Person entity that corresponds to them, so their name could be entered as plain text.

Comment (schmave) [reply modelsLibrary/app/models/RoleRecordMember.java:28]: Oh that's nice! I didn't notice that when I was testing it.

**Comment by schmave** [comment]
I'll deploy this tomorrow.

**Comment by schmave** [comment]
This is live on the site now!

**Comment by aaronbrowder** [comment]
Thank you! I'll check it out.

**Pull request merged**